draft-ietf-quic-tls.md

```diff
@@ -1991,6 +1991,16 @@ New QUIC versions SHOULD define a new salt value used in calculating initial
 secrets.
 
 
+## Randomness
+
+TLS relies on a cryptographically secure pseudorandom number generator
+(CSPRNG) {{!RFC4086}} to produce random or unpredictable values. In addition
+to the uses in TLS, a source of unpredictable values is used in QUIC for a
+variety of protocol elements, including stateless reset messages, stateless
+reset tokens, and PATH_CHALLENGE frames. An endpoint that employs a weak
+CSPRNG forfeits many of the security properties that QUIC provides.
```

Comment:
Is there a reason to be this opaque? I would just say that you SHOULD use a CSPRNG for most of these.

Note that some of these actually shouldn't be random, but are generated with a PRG.

Reply:
Yep. Agree with @ekr ; the spec should be less opaque on this matter. @martinthomson mentionned RFC 8937 in the linked issue , which I think would be great to mention in the QUIC-transport draft, as in the QUIC-tls one.

I would be happy to suggest a pull request to briefly explain the concern and connect it to the recent efforts in Post-Snowden crypto research  (rfc 8937 and linked papers). I think we somewhat need the Security Considerations section to connect the implementers with these research security questions currently being undertaken.

Reply:
There are cases where the CSPRNG is used and cases where a PRF or PRP is better (but a CSPRNG is acceptable, but maybe less efficient because it means remembering).  I wanted to avoid being concrete because this is an implementation choice often.  A recommendation in this case would be bad because a PRF/PRP is often a better choice.

Reply:
I agree with your point that some of these could be either generated with a PRF/PRP or a PRNG, but I think that this text simultaneously says too much and too little, because (for instance) you could have a strong PRNG and a weak PRF and a lot of these mechanisms would be weak. Constrariwise, you could have a weak PRNG used for QUIC but use a PRF for most of these and then you would be OK for those as long as you had a strong key.

So, I think I would say less:
"QUIC depends on the ability to generate secure random numbers, both directly for protocol values such as the connection ID, and transitively via TLS. See {{!RFC4086}} for guidance on secure random number generation."

Reply:
WFM, thanks.